packages/core/http/core-http-server/src/router/route.ts

```diff
@@ -321,6 +321,22 @@ export interface RouteConfigOptions<Method extends RouteMethod> {
    * @default false
    */
   httpResource?: boolean;
+
+  /**
+   * Based on the the ES API specification (see https://github.com/elastic/elasticsearch-specification)
+   * Kibana APIs can also specify some metadata about API availability.
+   *
+   * @remark intended to be used for informational purposes only.
+   *
+   */
+  availability?: {
+    /** @default stable */
```

Comment:
Yeah, I considered this too, but then realised our existing public routes would default to "experimental" too unless we explicitly set them all to "stable". So, in practice, we default to "stable" already today.

I'm not opposed to changing the default, and I tend to agree with you starting as "experimental" is better I just wanted to avoid a wider effort. We could do it as follow up work, happy to open an issue.

I wasn't too concerned about internal routes bc they are in principle "not available", but I think we can capture that in the doc comment.

Reply:
> Yeah, I considered this too, but then realised our existing public routes would default to "experimental" too unless we explicitly set them all to "stable".

We don't have that many public routes as we do internal. From a cost/benefit pov, changing public routes to `stable` is worth the effort of mitigating the risk of accidentally marking a route as stable when it's not.

Reply:
> I'm not opposed to changing the default, and I tend to agree with you starting as "experimental" is better I just wanted to avoid a wider effort. We could do it as follow up work, happy to open an issue.

Sounds good, LMK when you've created an issue.

Reply:
https://github.com/elastic/kibana/issues/197003 🙌🏻